lib/xcodeproj/workspace.rb

```diff
@@ -1,24 +1,48 @@
 require 'fileutils'
 require 'rexml/document'
 require 'xcodeproj/workspace/file_reference'
+require 'xcodeproj/workspace/group_reference'
 
 module Xcodeproj
   # Provides support for generating, reading and serializing Xcode Workspace
   # documents.
   #
   class Workspace
-    # @return [Array<String>] the paths of the projects contained in the
+    # @return [REXML::Document] the parsed XML model for the workspace contents
+    attr_reader :document
+    attr_reader :schemes
+
     # @return [Array<FileReference>] the paths of the projects contained in the
     #         workspace.
     #
-    attr_reader :file_references
-    attr_reader :schemes
+    def file_references
+      return [] unless @document
+      @document.get_elements('/Workspace//FileRef').map do |node|
+        FileReference.from_node(node)
+      end
+    end
 
-    # @param  [Array] file_references @see file_references
+    # @return [Array<GroupReference>] the groups contained in the workspace
     #
-    def initialize(*file_references)
-      @file_references = file_references.flatten
+    def group_references
+      return [] unless @document
+      @document.get_elements('/Workspace//Group').map do |node|
```

Comment:
The idea of this change is to work against the underlying xml to allow the document structure to remain intact when working with workspaces.